62_CAD/DrawResourcesFiller.h

```diff
@@ -205,12 +207,18 @@ struct DrawResourcesFiller
 		auto extents = msdfTextureArray->getCreationParameters().image->getCreationParameters().extent;
 		return uint32_t2(extents.width, extents.height);
 	}
+	uint32_t getMSDFMips() {
+		return msdfTextureArray->getCreationParameters().image->getCreationParameters().mipLevels;
+	}
+
+	// TODO: Return to protected after testing
+	uint32_t addMSDFTexture(std::function<core::smart_refctd_ptr<ICPUImage>()> createResourceIfEmpty, msdf_hash hash, SIntendedSubmitInfo& intendedNextSubmit);
 
 protected:
 	
 	struct TextureCopy
 	{
-		core::smart_refctd_ptr<ICPUBuffer> srcBuffer;
+		core::smart_refctd_ptr<ICPUImage> image;
 		uint64_t bufferOffset;
 		uint32_t3 imageExtent;
```

Comment:
I think you can remove these and for  MSDF Texture copies assert that the ICPUImage has same dimensions and mips as the draw resources filler IGPUImage (texture array)